dash/backend/src/modules/falco/dao/falco.dao.ts

```diff
@@ -192,11 +195,94 @@ export class FalcoDao {
         }
     }
 
-        async createFalcoLog(falcoLog: FalcoDto): Promise<number> {
+        async createFalcoLog(falcoLog: FalcoDto): Promise<FalcoDto> {
         const knex = await this.databaseService.getConnection();
         return knex
             .insert(instanceToPlain(falcoLog))
             .into('project_falco_logs')
-            .returning(['id']);
+            .returning(['*']);
+    }
+
+        async createFalcoSetting(clusterId: number, falcoSetting: FalcoSettingDto): Promise<any>{
+            const knex = await this.databaseService.getConnection();
+            const subquery = await knex.raw(`select exists( select 1 from falco_settings where cluster_id =? )`, [clusterId])
+                .then(res => res?.rows[0]?.exists);
+
+            if (!subquery){
+                const result = knex
+                    .insert(instanceToPlain(falcoSetting))
+                    .into('falco_settings')
+                    .returning(['*']);
+
+                return result;
+            }else {
+                const result = knex('falco_settings').update(instanceToPlain(falcoSetting)).where('cluster_id', clusterId);
+                return result;
+            }
+        }
+
+    async findFalcoSettingByClusterId(clusterId: number): Promise<FalcoSettingDto> {
+        const knex = await this.databaseService.getConnection();
+
+        // see if any settings matches the clusterid
+        const query = await knex.select().from('falco_settings').where('cluster_id', clusterId);
```

Comment:
Lets validate that this actually is returning a FalcoSettingDto. I think it may be returning an object with similar fields, but with underscores instead of camel case.

Reply:
I believe you can use the plainToInstance function to fix this bug. 

https://github.com/m9sweeper/m9sweeper/blob/main/dash/backend/src/modules/cluster/dao/cluster.dao.ts#L38

        const query = plainToInstance(FalcoSettingDto, await knex.select().from('falco_settings').where('cluster_id', clusterId))

Or something like that. I see we are doing that elsewhere in the application.